query/expression.go

```diff
@@ -23,6 +23,55 @@ import (
 	"github.com/square/metrics/query/aggregate"
 )
 
+const simultaneousFetchLimit = 4
```

Comment:
the one i dislike about `simultaneousFetchLimit` and more importantly `fetchTicketsQueue` being in file scope is that it's a global variable shared amongst everything. (as unlikely as it is) we can have multiple `api`s and `backend`s initialized and yet we are limiting the concurrency.

there are ways to limit this to be api-scoped or backend-scoped (i.e. enforce the limit per backend / api). it would be stored in some state (let's call this type `Environment`) for now, and would be shared amongst all the query execution.

Currently we don't have such an object - command has a member function with the signature `Execute(api.Backend, api.API)` and both Backend and API doesn't allow you to store such a state, but we can imagine passing a third parameter 

on the other hand, I realized that ticketing could've been implemented as a wrapper backend. a wrapper that takes an actual backend as an input, and just implements the ticker system in front of it. then we'll implement per-backend concurrency control without having a global variable.